client/ayon_silhouette/api/plugin.py

```diff
@@ -0,0 +1,191 @@
+from ayon_core.pipeline import (
+    Creator,
+    LoaderPlugin,
+    CreatedInstance,
+    AYON_INSTANCE_ID,
+    AVALON_INSTANCE_ID,
+    CreatorError,
+)
+from ayon_core.lib import BoolDef
+from . import lib
+
+import fx
+
+
+def cache_instance_data(shared_data):
+    """Cache instances for Creators shared data.
+
+    Create `silhouette_cached_instances` key when needed in shared data and
+    fill it with all collected instances from the scene under its
+    respective creator identifiers.
+
+    Args:
+        shared_data(Dict[str, Any]): Shared data.
+
+    """
+    if shared_data.get('silhouette_cached_instances') is None:
+        shared_data["silhouette_cached_instances"] = cache = {}
+
+        session = fx.activeSession()
+        if not session:
+            return
+
+        instance_ids = {AYON_INSTANCE_ID, AVALON_INSTANCE_ID}
```

Comment:
I think we don't need `AVALON_INSTANCE_ID` here at all.
 ```suggestion
        instance_ids = {AYON_INSTANCE_ID, AVALON_INSTANCE_ID}
```

Reply:
We might need to - or be very sure about setting the correct id everywhere where relevant: https://github.com/ynput/ayon-silhouette/pull/1#discussion_r1927218730